rust/worker/src/segment/distributed_hnsw_segment.rs

```diff
@@ -194,7 +194,18 @@ impl<'a> SegmentWriter<'a> for DistributedHNSWSegmentWriter {
                             }
                         },
                     };
-                    self.index.read().add(record.offset_id as usize, embedding);
+
+                    let mut index = self.index.upgradable_read();
```

Comment:
How expensive is resize operation? If it is expensive then does it mean that the tail latency of compaction time will increase because of this? Might be fine because compactions are not directly in the critical path?

Reply:
discussed offline - its a move + alloc of the entire data in hsnw